integrations/src/test/java/io/jenkins/plugins/casc/CredentialsReadmeTest.java

```diff
@@ -0,0 +1,146 @@
+            if (credentials instanceof BasicSSHUserPrivateKey) {
+                BasicSSHUserPrivateKey key = (BasicSSHUserPrivateKey) credentials;
+                assertThat(key.getPassphrase(), hasPlainText(PASSPHRASE));
+                assertThat(key.getPrivateKey(), equalTo(PRIVATE_KEY));
+                assertThat(key.getId(), anyOf(
+                    is("ssh_with_passphrase_provided"),
+                    is("ssh_with_passphrase_provided_via_file")));
+                assertThat(key.getUsername(), is("ssh_root"));
+                assertThat(key.getScope(), is(CredentialsScope.SYSTEM));
+            } else if (credentials instanceof UsernamePasswordCredentials) {
+                UsernamePasswordCredentials user = (UsernamePasswordCredentials) credentials;
+                assertThat(user.getUsername(), is("some-user"));
+                assertThat(user.getPassword(), hasPlainText(PASSWORD));
+                assertThat(user.getScope(), is(CredentialsScope.GLOBAL));
+            } else if (credentials instanceof StringCredentials) {
+                StringCredentials string = (StringCredentials) credentials;
+                assertThat(string.getId(), is("secret-text"));
+                assertThat(string.getSecret(), hasPlainText(TEXT));
+                assertThat(string.getScope(), is(CredentialsScope.GLOBAL));
+            } else if (credentials instanceof AWSCredentialsImpl) {
+                AWSCredentialsImpl aws = (AWSCredentialsImpl) credentials;
+                assertThat(aws.getId(), is("AWS"));
+                assertThat(aws.getAccessKey(), equalTo(ACCESS_KEY));
+                assertThat(aws.getSecretKey(), hasPlainText(SECRET_ACCESS_KEY));
+                assertThat(aws.getScope(), is(CredentialsScope.GLOBAL));
+            } else if (credentials instanceof FileCredentials) {
+                FileCredentials file = (FileCredentials) credentials;
+                assertThat(file.getId(), is("secret-file"));
+                assertThat(file.getFileName(), is(MYSECRETFILE_TXT));
+                String fileContent = IOUtils.toString(file.getContent(), StandardCharsets.UTF_8);
+                assertThat(fileContent, containsString("SUPER SECRET"));
+                assertThat(file.getScope(), is(CredentialsScope.GLOBAL));
+            } else if (credentials instanceof CertificateCredentialsImpl) {
+                CertificateCredentialsImpl cert = (CertificateCredentialsImpl) credentials;
+                assertThat(cert.getId(), is("secret-certificate"));
+                assertThat(cert.getPassword(), hasPlainText(PASSWORD));
+//                Path path = Paths.get(getClass().getResource("test.p12").toURI());
+//                SecretBytes uploadedKeystore = SecretBytes.fromBytes(Files.readAllBytes(path));
+//                assertThat(cert.getKeyStoreSource().getKeyStoreBytes(), is(uploadedKeystore.getPlainData()));
+                assertThat(cert.getScope(), is(CredentialsScope.GLOBAL));
```

Comment:
@jvz @timja This seems broken... Why would they differ?

Currently seeing: `DER input, Integer tag error`